src/main/java/com/microsoft/azure/util/AzureCredentials.java

```diff
@@ -511,20 +526,11 @@ public static KeyVaultCredentials getCredentialById(String credentialId) {
                         Collections.<DomainRequirement>emptyList()),
                 CredentialsMatchers.withId(credentialId));
         if (azureCredentials != null) {
-            return new KeyVaultClientAuthenticator(azureCredentials.data.getClientId(),
-                    azureCredentials.data.getClientSecret());
-        }
-
-        UsernamePasswordCredentials usernamePasswordCredentials = CredentialsMatchers.firstOrNull(
```

Comment:
@xuzhang3 Though the username/password credential was not well documented. Still have to make sure the users known this break change for lots of Azure related Jenkins plugins depending on this plugin.

Reply:
@gavinfish btw even secret string credentials were broken in the UI, it didn't seem possible to use them if you configured by creating a credentials

Reply:
Sure, totally make sense to remove it.